src/App.jsx

```diff
@@ -14,45 +16,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const buttons = [
+  {
+    name: 'Sort alphabetically',
+    color: 'is-info',
+  },
+  {
+    name: 'Sort by length',
+    color: 'is-success',
+  },
+  {
+    name: 'Reverse',
+    color: 'is-warning',
+  },
+];
+
+export const App = () => {
+  const [stateSort, setStateSort] = useState(null);
+  const [stateReverse, setStateReverse] = useState(null);
+  let goods = [...goodsFromServer];
+
+  switch (stateSort) {
+    case 'Sort alphabetically':
+      goods = goods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case 'Sort by length':
+      goods = goods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    case null:
+      goods = [...goodsFromServer];
+      break;
+    default:
+      goods = [...goodsFromServer];
+  }
+
+  if (stateReverse) {
+    goods = goods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+
+        {buttons.map((button) => {
+          const isSort = button.name !== stateSort;
+          const isReverse = button.name !== stateReverse;
+
+          return (
+            <button
+              key={button.name}
+              type="button"
+              className={classNames(
+                'button', button.color, { 'is-light': isSort && isReverse },
+              )}
+              onClick={() => {
+                if (button.name === 'Reverse') {
+                  if (stateReverse) {
+                    setStateReverse(null);
+                  } else {
+                    setStateReverse(button.name);
+                  }
+                } else {
+                  setStateSort(button.name);
+                }
+              }}
+            >
```

Comment:
it's not quite right to write such large functions in jsx. Please create a separate function for this